consensus/src/execution_ctx.rs

```diff
@@ -291,37 +285,38 @@ impl<'a, DB: Database, T: Operations + 'static> ExecutionCtx<'a, DB, T> {
                     error!("unable to re-publish a handled msg {:?}", err)
                 });
             }
-            // An error here means an phase considers this message as invalid.
-            // This could be due to failed verification, bad round/step.
+            // This is a message from future round or step.
+            // Save it in future_msgs to be processed when we reach
+            // same round/step.
+            Err(ConsensusError::FutureEvent) => {
+                trace!("future msg {:?}", msg);
+
+                self.future_msgs.lock().await.put_event(
+                    msg.header.round,
+                    msg.get_step(),
+                    msg,
+                );
+
+                return None;
+            }
+            Err(ConsensusError::PastEvent) => {
+                return self.process_past_events(msg).await;
+            }
+            Err(ConsensusError::InvalidValidation(QuorumType::NoQuorum)) => {
```

Comment:
This case should be removed by the next PR, since NoQuorum is indeed a valid result for Validation. Maybe add a reminder

Reply:
Indeed, I just added the specific case to log it as warn instead of Error

```
# before
{"timestamp":"2024-01-24T16:33:38.394082Z","level":"ERROR","message":"phase handler err: InvalidValidation(no_quorum)","target":"dusk_consensus::execution_ctx","spans":[{"iter":0,"name":"Ratification","pk":"pFPEcfxidLvwmFKR","round":17,"name":"main"}]}

# after
{"timestamp":"2024-01-29T10:12:36.322397Z","level":"WARN","event":"No quorum reached","iter":0,"target":"dusk_consensus::execution_ctx","spans":[{"iter":0,"name":"Ratification","pk":"pFPEcfxidLvwmFKR","round":1,"name":"main"}]}
```